### ISSUE TITLE
Create documentation and sample for sharing the same transitive platform dependency

### ISSUE THREAD

**Issue by xster** [flutter/flutter#52493]
Describe best practice for managing and sharing upstream dependencies like Crashalytics that's being used both by the platform and Flutter. 

cc @jmagman @RedBrogdon

**Comment by xster** [comment]
Part of the work is figuring out the matrix as well for subproject/aar/cocoapods/framework.

**Comment by xster** [comment]
I made a simple demo with https://github.com/xster/flutter-test/tree/master/t154_flutter_crashlytics and https://github.com/xster/flutter-test/tree/master/t154_android_crashlytics though it's still good to document it more systematically.

**Comment by xster** [comment]
Good to check on iOS first too but if there aren't too much specific coding needed to accomplish this, it could just be some additional text instructions on the flutter.dev website.

**Comment by xster** [comment]
Tagging @jmagman for the iOS part of the doc

**Comment by xster** [comment]
Tentatively assigning to @jmagman for the iOS part

**Comment by sfshaza2** [comment]
@jmagman, if this is a flutter.dev documentation issue, we can move the issue to the website repo. Is there more to it?

**Comment by jmagman** [comment]
@sfshaza2 yes I think it can be moved.  It's now tracking writing an iOS version of https://flutter.dev/docs/development/add-to-app/android/plugin-setup.
